R/data-combined.R

```diff
@@ -0,0 +1,867 @@
+      data <- dataSetToDataFrame(dataSets)
+
+      # if alternative names are provided, use them
+      data <- private$.renameDatasets(data, names)
+
+      # add column with grouping information and then
+      # add a column describing the type of data and then
+      # convert to tibble before returning the dataframe
+      data <- data %>%
+        private$.addGroupCol(groups) %>%
+        dplyr::mutate(dataType = "observed") %>%
+        dplyr::as_tibble()
+
+      return(data)
+    },
+
+    # extract dataframe from `SimulationResults` objects
+
+    .simResultsToDF = function(simulationResults,
+                               quantitiesOrPaths = NULL,
+                               population = NULL,
+                               individualIds = NULL,
+                               names = NULL,
+                               groups = NULL) {
+      # all input validation will take place in this function itself
+      data <- simulationResultsToDataFrame(
+        simulationResults = simulationResults,
+        quantitiesOrPaths = quantitiesOrPaths,
+        population        = population,
+        individualIds     = individualIds
+      )
+
+      # if names are not present, use paths as unique names
+      if (!"name" %in% names(data)) {
+        data <- dplyr::mutate(data, name = paths)
+      }
+
+      # if alternative names are provided, use them
+      if (!is.null(names)) {
+        data <- private$.renameDatasets(data, names) %>%
+          dplyr::mutate(
+            name = dplyr::case_when(
+              is.na(name) ~ paths,
+              TRUE ~ name
+            )
+          )
+      }
+
+      # add column with grouping information and then
+      #
+      # add a column describing the type of data;
+      # simulation results never have errors, but this leads to inconsistent
+      # output, so also add a column for `yErrorValues` outputs and then
+      #
+      # rename according to column naming conventions for `DataSet` and then
+      data <- data %>%
+        private$.addGroupCol(groups) %>%
+        dplyr::mutate(
+          dataType     = "simulated",
+          yErrorValues = NA_real_
+        ) %>%
+        dplyr::rename(
+          "xValues"    = "Time",
+          "xUnit"      = "TimeUnit",
+          "xDimension" = "TimeDimension",
+          "yValues"    = "simulationValues",
+          "yUnit"      = "unit",
+          "yDimension" = "dimension"
+        )
+
+      return(data)
+    },
+
+    ## dataframe modifiers ---------------------
+
+    # Add a new group column
+    #
+    # If no grouping is specified for a dataset, the group will be `NA`.
+    #
+    # While visualizing such data, the datset name can instead be used as a
+    # grouping variable, but this association will happen in the plotting
+    # function itself, and not in the output produced by the current object.
+
+    .addGroupCol = function(data, groups = NULL) {
+      if (!is.null(groups)) {
+        data <- data %>%
```

Comment:
This is the first line of code where I would really need to see comment to understand what it does. Grouping, nesting, ungrouping, mutate, unnest etc.. 
from the function comment, I am expecting this to add a column to data. but it does clearly something else
Also data is a df right? should we call it df? or dataFrame ? Thougths?

Reply:
Good point. 

I'have added comments to explain the _why_ and _how_ for what's going on in `.addGroupCol()` function.

I've also changed the return logic.